src/main/java/org/zone/region/flag/meta/service/ban/flag/BanFlag.java

```diff
@@ -0,0 +1,108 @@
+package org.zone.region.flag.meta.service.ban.flag;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.zone.region.flag.Flag;
+import org.zone.region.flag.FlagType;
+import org.zone.region.flag.FlagTypes;
+import org.zone.region.flag.meta.service.ban.BanInfo;
+import org.zone.region.flag.meta.service.ban.PermanentBanInfo;
+import org.zone.region.flag.meta.service.ban.TemporaryBanInfo;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Flag for banning players from a zone
+ */
+public class BanFlag implements Flag.Serializable {
+
+    /**
+     * Collection of ban info
+     */
+    public final Collection<BanInfo> banInfo = new HashSet<>();
```

Comment:
Dont need to show everyone our organs, private please

Reply:
Whoops I just cut an incision by mistake